llvm/docs/StackMaps.rst

```diff
@@ -338,14 +338,17 @@ format of this section follows:
     uint64 : PatchPoint ID
     uint32 : Instruction Offset
     uint16 : Reserved (record flags)
-    uint16 : NumLocations
-    Location[NumLocations] {
-      uint8  : Register | Direct | Indirect | Constant | ConstantIndex
-      uint8  : Reserved (expected to be 0)
-      uint16 : Location Size
-      uint16 : Dwarf RegNum
-      uint16 : Reserved (expected to be 0)
-      int32  : Offset or SmallConstant
+    uint16 : NumLiveVars
+    LiveVars[NumLiveVars] {
+      uint8  : NumLocations
```

Comment:
As I'm raising this PR, It's bothering me a bit that I only allocated a `u8` for the number of locations per live variable.

I hadn't realised that if you pass a huge array, say `my_array[2048]`, then each array element gets a location entry in the stackmap.

Ideally the stackmap format would use some kind of "compression" to avoid this for contiguous entries in memory, but that's out of scope for now.

I can go back and fix the `u8`, but it might be a fair amount of work, since it touches a lot of tests.

Reply:
Do we at least get a compile-time abort if such a situation happens?

Reply:
That is something I should certainly check!

Reply:
Fixed in [58277f5](https://github.com/ykjit/ykllvm/pull/40/commits/58277f539fdde3fa9b8906f409e7ffc4e6d19142)